### ISSUE TITLE
Amplify Flutter Auth 1.0 tutorials doesn't work right

### ISSUE THREAD

**Issue by Meerkov** [aws-amplify/amplify-flutter#3091]
### Description

In short, I gave up on Amplify Flutter and moved over to ReactJS because you can't even finish the tutorials without running into undocumented issues.

in long:
1) Complete the Flutter Amplify Todo tutorial (there are 3 or 4 different versions of this tutorial for Flutter Amplify with slightly different code samples, but any of them are fine).
2) Enabled CI/CD
3) Upgrade to the most recent version of Amplify Flutter that supports the Authenticator Module. Try to follow an official tutorial on adding Auth.
4) Get warnings that now your app only supports android SDK 33 (~5% of phones)
5) Get warning that the Flutter Authenticator module doesn't support web
6) Disable CI/CD because now that's broken
7) Build is broken because it can't find the environment variables for Google Client ID and Google Client Secret
8) Put those in Environment variables to fix the build, even though they already are in Amplify Studio's authenticator section
9) The tutorial tells you not use use Environment Variables and to instead use Environment Secrets. There is no tutorial on how to do that, and there is no explanation given as to why you wouldn't do it this way.

etc.

Basically using Amplify Flutter, even though it's v1.0, is like using a beta or alpha product.

Expected Result:
Amplify Flutter Auth module 1.0 would work out of the box on web/ios/android.

### Categories

- [ ] Analytics
- [ ] API (REST)
- [ ] API (GraphQL)
- [X] Auth
- [X] Authenticator
- [ ] DataStore
- [ ] Storage

### Platforms

- [X] iOS
- [X] Android
- [X] Web
- [ ] macOS
- [ ] Windows
- [ ] Linux

### Flutter Version

x

### Amplify Flutter Version

x

### Deployment Method

Amplify CLI

**Comment by Jordan-Nelson** [comment]
@Meerkov can you link to the specific tutorial you were following? Were you using Amplify Studio only, or were you using the CLI as well?

**Comment by Meerkov** [comment]
I don't have the exact error messages on hand because I nuked the project as a waste of time.

SDK 33 warning was presented in the CLI, saying that several libraries, including amplify_flutter_android have min SDK 33, and that you need to update the minSDK of the app.

Example of the multiple different similar tutorials
https://docs.amplify.aws/start/getting-started/installation/q/integration/flutter/
https://aws.amazon.com/getting-started/hands-on/build-flutter-mobile-app-part-one/
https://docs.amplify.aws/lib/auth/getting-started/q/platform/flutter/

**Comment by Jordan-Nelson** [comment]
Okay. Thanks for the links and additional info.

Regarding the min SDK, all Amplify Libraries support API level 24 and up on Android as of v1. If you can share the warning you saw I can try to look into this.

Regarding web support, all Amplify libraries except DataStore support web as of v1. One of the tutorials you linked was published prior to v1 being released and includes instructions to install versions on the library that did not support web. I can look into updating this. If you are sure you were using v1 or higher and can share the warning we can look into it.

We can take a look at the various guides to ensure that are up to date. Without more detailed info about the warnings you saw it may be hard to investigate this though and I understand you already deleted the project.

**Comment by abdallahshaban557** [comment]
Hi @Meerkov - would love to see where you have seen mentions of API Level 33? I believe we were chatting on Discord as well! Our documentation for the getting started guide has recently been updated https://docs.amplify.aws/start/getting-started/installation/q/integration/flutter/

We are in the midst of updating the hands on tutorial as well. It will completed soon. https://aws.amazon.com/getting-started/hands-on/build-flutter-mobile-app-part-one/

**Comment by Jordan-Nelson** [comment]
Hello @Meerkov - I am going to close this out since the [docs that were identified as outdated](https://aws.amazon.com/getting-started/hands-on/build-flutter-mobile-app-part-one/) have been updated.
